voxblox/src/integrator/esdf_integrator.cc

```diff
@@ -370,12 +366,17 @@ void EsdfIntegrator::processOpenSet() {
       continue;
     }
 
-    // Get the global indices of neighbors.
-    neighbor_tools_.getNeighborsByGlobalIndex(
-        global_index, Connectivity::kTwentySix, &neighbors);
+    Neighborhood<Connectivity::kTwentySix>::getFromGlobalIndex(
+        global_index, &neigbor_indices);
 
     // Go through the neighbors and see if we can update any of them.
-    for (const GlobalIndex& neighbor_index : neighbors) {
+    for (unsigned int idx = 0u; idx < neigbor_indices.cols(); ++idx) {
+      const GlobalIndex& neighbor_index = neigbor_indices.col(idx);
+      const SignedIndex& direction =
+          NeighborhoodLookupTables::kOffsets.col(idx);
+      FloatingPoint distance =
+          NeighborhoodLookupTables::kDistances[idx] * voxel_size_;
```

Comment:
If we want to squeeze even a bit more out of this, we could precompute the metric distances based on the voxel size, we could e.g. store it in the esdf integrator as a member when it is initialized.